### DISCUSSION TITLE
Override or run custom command in an action service container

### DISCUSSION THREAD

**Discussion by AssisrMatheus** [community/community#25276] [Actions]
<p>I am running a workflow on each PR to test my application with cypress. This workflow raises two <a href="https://docs.github.com/en/actions/using-containerized-services/about-service-containers" rel="noopener nofollow ugc">service containers</a>, one for the database, and another for the backend.</p>
<p>In one of the steps. I need to run the seed inside the backend service. To get the database ready for testing. And this is only needed for testing, so this command isn’t inside the <code>Dockerfile</code> itself</p>
<p>I tried using the <code>--entrypoint</code> option in the backend service, to make  it run an specific script in <code>package.json</code> that does what I need, but then I get another error:</p>
<pre><code class="lang-auto">/usr/bin/docker start 49fef80427d4bae51d14f00eeea4b1039ce3c11037fe6ef1afebc37f9f9cc498
  Error response from daemon: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "npm run test:frontend": executable file not found in $PATH: unknown
  Error: failed to start containers: 49fef80427d4bae51d14f00eeea4b1039ce3c11037fe6ef1afebc37f9f9cc498
  Error: Docker start fail with exit code 1
</code></pre>
<p><code>test:frontend</code> just runs the commands that were installed from npm, and the base image I’m using is a node image, so I am also sure that NPM is installed</p>
<p>I tried using a step to run a docker exec command, to run the seed within the running service. But I don’t think I have access to that. And had its other set of issues: <a href="https://github.community/t/docker-not-found-on-run-step/232003" class="inline-onebox">"docker: not found" on run step</a></p>
<p>How can I override the command of fix the error I’m having?</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="AssisrMatheus" data-post="1" data-topic="232637">
<div class="title">
<div class="quote-controls"></div>
 AssisrMatheus:</div>
<blockquote>
<p><code>  Error response from daemon: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "npm run test:frontend": executable file not found in $PATH: unknown</code></p>
</blockquote>
</aside>
<p>That message looks like the command is formatted in such a way that Docker assumes you want to run a binary called <code>npm run test:frontend</code>, not <code>npm</code> with the rest being arguments.</p>
